kerchunk/grib2.py

```diff
@@ -190,12 +246,12 @@ def scan_grib(
                 coord2 = {"latitude": "latitudes", "longitude": "longitudes"}.get(
                     coord, coord
                 )
-                if coord2 in m:
-                    x = m[coord2]
-                else:
+                x = m.get(coord2)
+                if x is None:
                     continue
```

Comment:
This is a large improvement because even `__contains__` reads out "coord2".

Unfortunately, there doesn't seem to be any way to get a list of valid keys for a message (see comments above).

Even cfgrib just tries indexing and catches the error:
https://github.com/ecmwf/cfgrib/blob/f9106c8bc6b18238aa1e7d54100e6e6364d62ffe/cfgrib/dataset.py#L470-L479